### PR TITLE
protectli_vp46xx/firmware-update.md: add information about versions newer than 1.0.18

### PR DESCRIPTION
**Pull request by sulewskiprzemyslaw**
Signed-off-by: Przemyslaw Sulewski <przemyslaw.sulewski@3mdeb.com>

### CODE REVIEW

File: docs/variants/protectli_vp46xx/firmware-update.md [hunk 1]
```diff
@@ -9,10 +9,11 @@ firmware version.
 For simplicity we recommend using
 [Dasharo Tools Suite](../../../common-coreboot-docs/dasharo_tools_suite).
 
-## Updating to Dasharo v1.0.18
+## Updating to Dasharo v1.0.18 or newer versions
```
Comment (miczyg1) [inline docs/variants/protectli_vp46xx/firmware-update.md:12]: if one updates from v1.0.18 to v1.0.19 flashing only WP_RO and RW_SECTION_A is sufficient. BUt if v1.0.19 is goign to be the first Dasahro version being installed, full flashing is needed

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: docs/variants/protectli_vp46xx/firmware-update.md [hunk 2]
```diff
@@ -9,12 +9,23 @@ firmware version.
...
 [Dasharo Tools Suite](../../../common-coreboot-docs/dasharo_tools_suite).
 
-## Updating to Dasharo v1.0.18 or newer versions
+## Updating to Dasharo v1.0.18 or v1.0.19
 
 From v1.0.18 Dasharo firmware is rebased on the more up-to-date revision of
-coreboot. This results in a requirement to flash the whole image as described in
+coreboot.
+
+If the current version of the firmware on the device is older than v1.0.18 the
```
Comment (miczyg1) [inline docs/variants/protectli_vp46xx/firmware-update.md:17]:
```suggestion
If the current version of the firmware on the device is older than v1.0.18 or
you are migrating from proprietary firmawre the
```

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: docs/variants/protectli_vp46xx/firmware-update.md [hunk 3]
```diff
@@ -9,10 +9,22 @@ firmware version.
...
-and it requires to flash whole image as described in [Initial Deployment](initial-deployment.md).
+From v1.0.18 Dasharo firmware is rebased on the more up-to-date revision of
+coreboot.
+
+If the current version of the firmware on the device is older than v1.0.18 or
+you are migrating from proprietary firmware the whole flash chip should be
+flashed as described in [Initial Deployment](initial-deployment.md).
+
+If the current version of the firmware on the device is v1.0.18 and it should
+be updated to v1.0.19, only the `WP_RO` and `RW_SECTION_A` should be flashed.
```
Comment (miczyg1) [inline docs/variants/protectli_vp46xx/firmware-update.md:22]: Forgot to mention that the Dasahro BIOS Boot medium lock must be disabled beforehand...

File: docs/variants/protectli_vp46xx/firmware-update.md [hunk 4]
```diff
@@ -9,10 +9,22 @@ firmware version.
...
-## Updating to Dasharo v1.0.18
+## Updating to Dasharo v1.0.18 or v1.0.19
 
-v1.0.18 is a refreshed release, rebased on more up-to-date revision of coreboot
-and it requires to flash whole image as described in [Initial Deployment](initial-deployment.md).
+From v1.0.18 Dasharo firmware is rebased on the more up-to-date revision of
+coreboot.
+
+If the current version of the firmware on the device is older than v1.0.18 or
+you are migrating from proprietary firmware the whole flash chip should be
```
Comment (miczyg1) [inline docs/variants/protectli_vp46xx/firmware-update.md:18]: Proprietary firmware also need to disable `BIOS Lock` first

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: docs/variants/protectli_vp46xx/firmware-update.md [hunk 5]
```diff
@@ -9,10 +9,34 @@ firmware version.
 For simplicity we recommend using
 [Dasharo Tools Suite](../../../common-coreboot-docs/dasharo_tools_suite).
 
-## Updating to Dasharo v1.0.18
+Before starting the update procedure be sure to disable Dasahro BIOS Boot medium
```
Comment (miczyg1) [inline docs/variants/protectli_vp46xx/firmware-update.md:12]:
```suggestion
Before starting the update procedure be sure to disable Dasharo BIOS Boot medium
```

File: docs/variants/protectli_vp46xx/firmware-update.md [hunk 6]
```diff
@@ -9,10 +9,34 @@ firmware version.
...
+lock:
```
Comment (miczyg1) [inline docs/variants/protectli_vp46xx/firmware-update.md:13]:
```suggestion
lock (if you already installed Dasharo and want to update):
```

File: docs/variants/protectli_vp46xx/firmware-update.md [hunk 7]
```diff
@@ -9,10 +9,34 @@ firmware version.
...
-and it requires to flash whole image as described in [Initial Deployment](initial-deployment.md).
+1. Power on the device.
+1. While the device is booting, hold the `DELETE` key to enter the UEFI Setup
+    Menu.
+1. Enter the `Dasharo System Features` menu using the arrow keys and Enter.
+1. Enter the `Dasharo Security Options` submenu.
+1. Verify the state of the `Lock the BIOS boot medium` option - if the option
+    is chosen, press `Space` and then `F10` to save the changes.
+1. Reboot the device to properly apply the changes.
+
```
Comment (miczyg1) [inline docs/variants/protectli_vp46xx/firmware-update.md:23]: We need the procedure for AMI BIOS too.

**Review by miczyg1 — CHANGES_REQUESTED** [review]

**Comment by sulewskiprzemyslaw** [comment]
@miczyg1 in the first section I just added a reference to the Initial deployment documentation ("If your device is currently flashed with the proprietary firmware please refer to Initial deployment documentation.").

In my opinion, that reference resolves the problems described in the last requests.

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**